File: Containers/src/WebApp/Services/BasketItem.cs
namespace eShop.WebApp.Services;

public class BasketItem
{
    public required string Id { get; set; }
    public int ProductId { get; set; }
    public required string ProductName { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal OldUnitPrice { get; set; }
    public int Quantity { get; set; }
}

File: src-java/eshop-integration-events/src/main/java/com/eshop/integrationevents/BasketItem.java
package com.eshop.integrationevents;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.util.Objects;

public record BasketItem(
        @JsonProperty("Id") String id,
        @JsonProperty("ProductId") int productId,
        @JsonProperty("ProductName") String productName,
        @JsonProperty("UnitPrice") BigDecimal unitPrice,
        @JsonProperty("OldUnitPrice") BigDecimal oldUnitPrice,
        @JsonProperty("Quantity") int quantity,
        @JsonProperty("PictureUrl") String pictureUrl) {

    public BasketItem {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(productName, "productName");
        Objects.requireNonNull(unitPrice, "unitPrice");
        Objects.requireNonNull(oldUnitPrice, "oldUnitPrice");
        if (quantity < 0) {
            throw new IllegalArgumentException("quantity must not be negative");
        }
    }
}
